frontend/staff-dashboard/src/hooks/drfDataProvider.ts

```diff
@@ -0,0 +1,172 @@
+
+const axiosInterface = axios.create();
+
+axiosInterface.interceptors.request.use((config: any) => {
+  console.log('we get use!')
+  console.log(config);
+
+  let token = sessionStorage.getItem(`oidc.user:${OIDC_CONFIG.authority}:${OIDC_CONFIG.client_id}`);
+
+  if (token !== null) {
+    token = JSON.parse(token).access_token;
+    console.log(`the token is ${token}`)
+    config.headers.Authorization = `Bearer ${token}`;
+  }
+
+  return config;
+}, (error: any) => Promise.reject(error));
+
+const generateSort = (sort?: CrudSorting) => {
+    if (sort && sort.length > 0) {
+        const _sort: string[] = [];
+        const _order: string[] = [];
+
+        sort.map((item) => {
+            _sort.push(item.field);
+            _order.push(item.order);
+        });
+
+        return {
+            _sort,
+            _order,
+        };
+    }
+
+    return;
+};
+
+const mapOperator = (operator: CrudOperators): string => {
+    switch (operator) {
+        case "ne":
+        case "gte":
+        case "lte":
+            return `_${operator}`;
+        case "contains":
+            return "_like";
+        case "eq":
+        default:
+            return "";
+    }
+};
+
+const generateFilter = (filters?: CrudFilters) => {
+    const queryFilters: { [key: string]: string } = {};
+    if (filters) {
+        filters.map((filter) => {
+            if (filter.operator !== "or") {
+                const { field, operator, value } = filter;
+
+                if (field === "q") {
+                    queryFilters[field] = value;
+                    return;
+                }
+
+                const mappedOperator = mapOperator(operator);
+                queryFilters[`${field}${mappedOperator}`] = value;
+            }
+        });
+    }
+
+    return queryFilters;
+};
+
+const DrfDataProvider = (
+    apiUrl: string,
+    httpClient: AxiosInstance = axiosInterface
+): DataProvider => {
+    const simpleDataProvider = dataProvider(apiUrl, httpClient);
+
+    simpleDataProvider.getList = async ({ resource, pagination, filters, sort }) => {
+        console.log("cheeses");
+        const url = `${apiUrl}/${resource}/`;
+
+        // pagination
+        const current = pagination?.current || 1;
+        const pageSize = pagination?.pageSize || 10;
+
+        const queryFilters = generateFilter(filters);
+
+        const query: {
+            o: number;
+            l: number;
+            _sort?: string;
+            _order?: string;
+        } = {
+            o: (current - 1) * pageSize,
+            l: pageSize,
+        };
+
+        const generatedSort = generateSort(sort);
+        if (generatedSort) {
+            const { _sort, _order } = generatedSort;
+            query._sort = _sort.join(",");
+            query._order = _order.join(",");
+        }
+
+        console.log(query);
+
+        const uri = `${url}?${stringify(query)}&${stringify(queryFilters)}`;
+
+        console.log(`uri is ${uri}`);
```

Comment:
You can condense all of this down to a few lines:

```typescript
const query = {
    ...generateFilter(filters),
    ...generatePagination(pagination),  // note this should be implemented as a refactor
    ...generateSort(sort), // this should now return {} instead of undefined if no sort
}
const uri = `${url}?${stringify(query)}`;
```